packages/ssi-sdk-core/src/utils/image.ts

```diff
@@ -1,25 +1,67 @@
 import fetch from 'cross-fetch'
-import sizeOf from 'image-size'
-import { ISizeCalculationResult } from 'image-size/dist/types/interface'
+import { imageSize } from 'image-size'
 import { IImageDimensions, IImageResource } from '../types'
 
+type SizeCalculationResult = {
+  width?: number
+  height?: number
+  orientation?: number
+  type?: string
+}
+
+const uint8ArrayToString = (uint8Array: Uint8Array): string => {
+  const decoder = new TextDecoder('utf-8')
+  return decoder.decode(uint8Array)
+}
+
+// TODO: here we're handling svg separately, remove this section when image-size starts supporting it in version 2
+const isSvg = (uint8Array: Uint8Array): boolean => {
+  const text = uint8ArrayToString(uint8Array)
+  const normalizedText = text.trim().toLowerCase()
+  return normalizedText.startsWith('<svg') || normalizedText.startsWith('<?xml')
+}
+
+const getSvgDimensions = (uint8Array: Uint8Array) => {
+  const svgContent = uint8ArrayToString(uint8Array)
+  const widthMatch = svgContent.match(/width="([^"]+)"/)
+  const heightMatch = svgContent.match(/height="([^"]+)"/)
+  const viewBoxMatch = svgContent.match(/viewBox="([^"]+)"/)
+
+  let width, height
+
+  if (widthMatch) {
+    width = widthMatch[1]
+  }
+
+  if (heightMatch) {
+    height = heightMatch[1]
+  }
+
+  if ((!width || !height) && viewBoxMatch) {
+    const parts = viewBoxMatch[1].split(' ').map(Number)
+    if (!width && parts.length === 4) {
+      width = parts[2].toString()
```

Comment:
Why a tostring on something that is mapped to a number?

Reply:
changed this, width/height is a number now